### PR TITLE
[TTAHUB-3096] Include inactive grants that fall within 60 day inactivation date

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This change ensures that we only show inactive grants if the inactivation date is 60 days from either todays date for new reports. Or 60 days from the start date if an existing report. In Jon's previous PR he ensures that any grant that was linked via the activity recipient's table will still show in the grant list regardless of date.

## How to test

- Review the code.
- Make sure for a new report we show inactive grants within 60 days from today's date.
- Make sure for existing reports with and without a start date we show inactive grants respectively.
- Make sure Jon's fix in the previous PR still works as expected (see above description).

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3096

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: src/services/activityReports.js [hunk 1]
```diff
@@ -1043,6 +1042,21 @@ export async function setStatus(report, status) {
...
+    where: {
+      id: activityReportId,
+    },
+  }) : null;
+
+  // If we have a valid startDate subtract 61 days from it.
+  let startDate = null;
+  if (reportWithStartDate && reportWithStartDate.startDate) {
+    startDate = moment(reportWithStartDate.startDate).subtract(61, 'days').format('YYYY-MM-DD');
+  }
```
Comment (GarrettEHill) [inline src/services/activityReports.js:1058]: why not do this all in sql 
```suggestion
    const reportWithStartDate = activityReportId ? await ActivityReport.findOne({
      attributes: [
        'startDate',
        [sequelize.literal(`"startDate" - INTERVAL '61 days'`), 'adjustedStartDate']
      ],
      where: {
        id: activityReportId,
      },
    }) : null;

  // If we have a valid startDate subtract 61 days from it.
  const startDate = reportWithStartDate && reportWithStartDate.startDate
      ? reportWithStartDate.get('adjustedStartDate')
      :  null;
```

Comment (AdamAdHocTeam) [reply src/services/activityReports.js:1058]: I did initially we don't have a link to that report. Right now reports are joined through ActivityRecipients table we need to handle the case were a inactive grant is NOT linked to an AR.

File: src/services/activityReports.js [hunk 2]
```diff
@@ -1080,7 +1094,13 @@ export async function possibleRecipients(regionId, activityReportId = null) {
       '$grants.regionId$': regionId,
       [Op.or]: [
         { '$grants.status$': 'Active' },
-        { '$grants->activityRecipients.activityReportId$': activityReportId },
+        { ...(activityReportId ? { '$grants.activityRecipients.activityReportId$': activityReportId } : {}) },
+        {
+          '$grants.inactivationDate$': {
+            [Op.gte]:
+              sequelize.literal(`${!startDate ? `date_trunc('day', NOW()) - interval '61 days'` : `'${startDate}'`}`),
```
Comment (GarrettEHill) [inline src/services/activityReports.js:1101]:
```suggestion
           [Op.gte]: sequelize.literal(`
        CASE
          WHEN ${activityReportId ? 'true' : 'false'}
          THEN (SELECT COALESCE("startDate", NOW()) - INTERVAL '61 days' FROM "ActivityReports" WHERE "id" = ${activityReportId})
          ELSE date_trunc('day', NOW()) - interval '61 days'
        END
      `)
```

File: src/services/activityReports.js (hunk 1 above)
Comment (GarrettEHill) [reply src/services/activityReports.js:1058]: just remove this whole block

**Comment by hardwarehuman** [comment]
Your link in the PR description "Issue(s)" section is to the wrong Jira

File: src/services/activityReports.js [hunk 3]
```diff
@@ -1043,6 +1042,21 @@ export async function setStatus(report, status) {
...
+    attributes: ['startDate'],
+    where: {
+      id: activityReportId,
+    },
+  }) : null;
+
+  // If we have a valid startDate subtract 61 days from it.
+  let startDate = null;
+  if (reportWithStartDate && reportWithStartDate.startDate) {
+    startDate = moment(reportWithStartDate.startDate).subtract(61, 'days').format('YYYY-MM-DD');
```
Comment (hardwarehuman) [inline src/services/activityReports.js:1057]: I don't think this is correct business logic. If we have the actual startDate for the AR, then we just want to get the grants active on that date. It's when we do _not_ have a startDate that we hunt back 61 days from the day the AR record is opened

Comment (AdamAdHocTeam) [reply src/services/activityReports.js]: This code has been removed. We just added a select in the case. See latest updtes.

**Review by hardwarehuman — CHANGES_REQUESTED** [review]
I believe the 60/61 day adjustment should only apply to ARs that don't have a startDate and thus could easily have been in service of a grant that inactivated between when the activity occurred and when the AR is written. The current logic would help people create ARs attached to grants that expired before the actual activity itself took place. I don't think that was what the Jira meant.

File: src/services/activityReports.js [hunk 4]
```diff
@@ -1080,7 +1080,18 @@ export async function possibleRecipients(regionId, activityReportId = null) {
...
       [Op.or]: [
         { '$grants.status$': 'Active' },
-        { '$grants->activityRecipients.activityReportId$': activityReportId },
+        { ...(activityReportId ? { '$grants.activityRecipients.activityReportId$': activityReportId } : {}) },
+        {
+          '$grants.inactivationDate$': {
+            [Op.gte]: sequelize.literal(`
+          CASE
+            WHEN ${activityReportId ? 'true' : 'false'}
+            THEN (SELECT COALESCE("startDate", NOW()) - INTERVAL '${inactiveDayDuration} days' FROM "ActivityReports" WHERE "id" = ${activityReportId})
```
[the remark this answers is not in the document]
Comment (GarrettEHill) [reply src/services/activityReports.js:1089]: if start date is null then it acts just like the else case with the original code, yours does not do the date adjustment at all if start date is populated.

[the remark this answers is not in the document]
Comment (GarrettEHill) [reply src/services/activityReports.js:1089]: I guess I see your logic, then yea. yours might be a better solution.

**Review by hardwarehuman — APPROVED** [review]
This looks right to me now

**Review by GarrettEHill — APPROVED** [review]

**Review by nvms — APPROVED** [review]
I think I tested all the cases. Looks good!

**Pull request merged**